app/models/patient.rb

```diff
@@ -162,6 +164,28 @@ def event_params
     }
   end
 
+  def confirm_unique_phone_number
+    ##
+    # This method is preferred over Rail's built-in uniqueness validator
+    # so that case managers get a meaningful error message when a patient
+    # exists on a different line than the one the volunteer is serving.
+    #
+    # See https://github.com/DCAFEngineering/dcaf_case_management/issues/825
+    ##
+    already_taken_patient = Patient.where(primary_phone: primary_phone)
```

Comment:
Can change this `where` to a `find`, right? Since there will either be one or zero. That'll save us from having to do the index lookup on L178.

Reply:
`find` works by ids. There is `find_by`, which looks like this:

```ruby
Patient.find_by(primary_phone: primary_phone)
```
[Mongoid docs for #find_by](https://docs.mongodb.com/mongoid/6.3/api/Mongoid/Findable.html#find_by-instance_method)

Would that work? I dunno databases too well

Reply:
`find_by` has an tradeoff: code readability. `find_by` will throw an error when no match is found. 

There are two ways to handle this. Option 1 is something like this:
```ruby
begin
  already_taken_patient = Patient.find_by(primary_phone: primary_phone)
rescue
  return  // terminate the function since number is verified to be unique
end
```

(Syntax might be off.)

Option 2 is configuring `Mongoid.raise_not_found_error` to be `false`.

Oh, and option 3 is keeping the `where` syntax.

Reply:
I've changed it to `Patient.where(primary_phone: primary_phone).first`, and rewrote the rest to match that.

Also, I'm just realizing as I type this that "index lookup on L178" is not a fancy obscure database perf issue, but just you wanting me to avoid `already_taken_patient[1]`.